src/components/NavButtonWithTooltip/index.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useState } from "react";
+import { Tooltip, TooltipProps } from "antd";
+
+import {
+    NAV_BAR_TOOLTIP_OFFSET,
+    TOOLTIP_COLOR,
+    TOOLTIP_DELAY,
+} from "../../constants";
+import NavButton, { NavButtonProps } from "../NavButton";
+
+interface TooltipText {
+    default: string;
+    disabled?: string;
+}
+interface NavButtonWithTooltipProps extends NavButtonProps {
+    tooltipText?: TooltipText;
+    tooltipPlacement?: TooltipProps["placement"];
```

Comment:
how does this work? It looks clunky to use the brackets and string literal.  Does ts let you do something like `typeof TooltipProps.placement` or can you specify it more explicitly?

Reply:
Hmm, that `typeof` version doesn't work directly.

Antd actually exports `TooltipPlacement` as a type but not from their top level I have to import from `antd/lib/tooltip` which is maybe not the most maintainable?

Alternately can just redeclare it with all the string literals in our own types file and import it, that might be the best way imo